quinn-proto/src/endpoint.rs

```diff
@@ -1193,6 +1193,17 @@ impl Incoming {
     pub fn remote_address_validated(&self) -> bool {
         self.retry_src_cid.is_some()
     }
+
+    /// Ignore this incoming connection attempt without cleaning up the corresponding data
+    /// structures in [`Endpoint`]
+    ///
+    /// Usually one should use [`Endpoint::ignore`] instead, as allowing endpoint to clean up its
+    /// state tracking the incoming connection is necessary to prevent memory leaks. This method
+    /// drops the `Incoming` without doing that cleanup but also without logging a warning, and is
+    /// designed for use when both the `Incoming` and the [`Endpoint`] are being dropped.
+    pub fn ignore_no_cleanup(self) {
```

Comment:
Rather than complicating the API, maybe just use `ignore`? The extra work should be pretty trivial.

Reply:
Good idea